beater/config_test.go

```diff
@@ -304,3 +278,94 @@ func TestPipeline(t *testing.T) {
 			fmt.Sprintf("<%v> shouldOverwrite() expected %v", idx, test.overwrite))
 	}
 }
+
+func TestTLSSettings(t *testing.T) {
+
+	t.Run("ClientAuthentication", func(t *testing.T) {
+		for name, tc := range map[string]struct {
+			config map[string]interface{}
+
+			tls *tlscommon.ServerConfig
+		}{
+			"Defaults": {
+				config: map[string]interface{}{"ssl": nil},
+				tls:    &tlscommon.ServerConfig{ClientAuth: 3},
+			},
+			"ConfiguredToRequired": {
+				config: map[string]interface{}{"ssl": map[string]interface{}{"client_authentication": "required"}},
+				tls:    &tlscommon.ServerConfig{ClientAuth: 4},
+			},
+			"ConfiguredToNone": {
+				config: map[string]interface{}{"ssl": map[string]interface{}{"client_authentication": "none"}},
+				tls:    &tlscommon.ServerConfig{ClientAuth: 0},
+			},
+			"DefaultRequiredByCA": {
+				config: map[string]interface{}{"ssl": map[string]interface{}{
+					"certificate_authorities": []string{"./path"}}},
+				tls: &tlscommon.ServerConfig{ClientAuth: 4},
+			},
+			"ConfiguredWithCA": {
+				config: map[string]interface{}{"ssl": map[string]interface{}{
+					"certificate_authorities": []string{"./path"}, "client_authentication": "none"}},
+				tls: &tlscommon.ServerConfig{ClientAuth: 0},
```

Comment:
This `ClientAuth` comes from [these constants](https://golang.org/pkg/crypto/tls/#ClientAuthType) right?  Can we use those instead?

Reply:
Actually [tlsClientAuth](https://github.com/elastic/beats/blob/master/libbeat/common/transport/tlscommon/types.go#L122) is an unexported type from the libbeat package. I cannot use go's constants here.

Reply:
I figured it was something strange like that, thanks.